frontend/src/components/IstioStatus/IstioStatus.tsx

```diff
@@ -145,6 +147,22 @@ export const IstioStatusComponent: React.FC<Props> = (props: Props) => {
     fetchStatus();
   }, [lastRefreshAt, fetchStatus]);
 
+  // Overview/Mesh pages do not have Namespaces selection, so separate fetchStatus call is necessary for them
+  React.useEffect(() => {
+    const currentPath = location.pathname;
+    const isOverviewOrMesh = currentPath.includes('/overview') || currentPath.includes('/mesh');
+    const pathChanged = currentPathRef.current !== currentPath;
+
+    if (pathChanged) {
+      currentPathRef.current = currentPath;
+
+      if (isOverviewOrMesh) {
+        fetchStatus();
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [location.pathname, fetchStatus]);
+
```

Comment:
```suggestion
  React.useEffect(() => {
    const currentPath = location.pathname;
    const pathChanged = currentPathRef.current !== currentPath;

    if (pathChanged) {
      currentPathRef.current = currentPath;
      fetchStatus();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [location.pathname, fetchStatus]);

```

Good investigation realizing that the status was tied to namespaces. That is pretty confusing.  I'm not sure the old code tying IstioStatus to namespaces even makes sense anymore, it may be unnecessary.  I think we could remove the redux connections to `namespaces` and `refreshNamespaces` and simplify the code above as suggested. That would remove this conditional logic that needs to know what pages we're dealing with, and just be clear that we want to update status whenever any page location changes.

Can you try making that change and see if it works? I did a quick test and it looked ok to me.

Reply:
Here , when removing `isOverviewOrMesh`, then the `/status` call will happen twice to all other paged besides Overview and Mesh, one with this `useEffect` and the second with `useEffect` above which is triggered by Namespace dropdown.
I tried to merge two `useEffect` functions into one, but what made the code over-complicated and hard to read. So I will suggest to leave newly added 'useEffect' as it is and add TODO comment for future Namespaces potential redesign.

Reply:
> when removing isOverviewOrMesh, then the /status call will happen twice t

Right, you have to also "_remove the redux connections to namespaces and refreshNamespaces_". Meaning, you need to remove the "tie" to namespace changes.

So, I think you can remove these lines (and associated props):
https://github.com/kiali/kiali/blob/6b4d40ee19e45ca1f787e3864e8ae9c958c2f0ad/frontend/src/components/IstioStatus/IstioStatus.tsx#L302
https://github.com/kiali/kiali/blob/6b4d40ee19e45ca1f787e3864e8ae9c958c2f0ad/frontend/src/components/IstioStatus/IstioStatus.tsx#L309-L311